packages/fast-components-react-msft/src/flyout/flyout.tsx

```diff
@@ -0,0 +1,202 @@
+import React from "react";
+import ReactDOM from "react-dom";
+import { get } from "lodash-es";
+import { canUseDOM } from "exenv-es6";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    AxisPositioningMode,
+    ViewportPositioner,
+    ViewportPositionerClassNameContract,
+    ViewportPositionerHorizontalPosition,
+    ViewportPositionerVerticalPosition,
+} from "@microsoft/fast-components-react-base";
+import {
+    FlyoutAxisPositioningMode,
+    FlyoutHandledProps,
+    FlyoutHorizontalPosition,
+    FlyoutManagedClasses,
+    FlyoutProps,
+    FlyoutUnhandledProps,
+    FlyoutVerticalPosition,
+} from "./flyout.props";
+import {
+    FlyoutClassNameContract,
+    ManagedClasses,
+} from "@microsoft/fast-components-class-name-contracts-msft";
+import { DisplayNamePrefix } from "../utilities";
+
+class Flyout extends Foundation<FlyoutHandledProps, FlyoutUnhandledProps, {}> {
+    public static displayName: string = `${DisplayNamePrefix}Flyout`;
+
+    public static defaultProps: Partial<FlyoutProps> = {
+        visible: false,
+        horizontalPositioningMode: FlyoutAxisPositioningMode.uncontrolled,
+        defaultHorizontalPosition: FlyoutHorizontalPosition.uncontrolled,
+        verticalPositioningMode: FlyoutAxisPositioningMode.adjacent,
+        defaultVerticalPosition: FlyoutVerticalPosition.bottom,
+        horizontalAlwaysInView: false,
+        verticalAlwaysInView: false,
+        fixedAfterInitialPlacement: false,
+        height: "128px",
+        width: "280px",
+    };
+
+    protected handledProps: HandledProps<Required<FlyoutHandledProps>> = {
+        anchor: void 0,
+        children: void 0,
+        defaultHorizontalPosition: void 0,
+        defaultVerticalPosition: void 0,
+        describedBy: void 0,
+        visible: void 0,
+        fixedAfterInitialPlacement: void 0,
+        height: void 0,
+        horizontalAlwaysInView: void 0,
+        horizontalLockToDefault: void 0,
+        horizontalPositioningMode: void 0,
+        horizontalThreshold: void 0,
+        label: void 0,
+        labelledBy: void 0,
+        managedClasses: void 0,
+        onDismiss: void 0,
+        verticalAlwaysInView: void 0,
+        verticalLockToDefault: void 0,
+        verticalPositioningMode: void 0,
+        verticalThreshold: void 0,
+        viewport: void 0,
+        width: void 0,
+    };
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <ViewportPositioner
+                {...this.unhandledProps()}
+                role="dialog"
+                aria-label={this.props.label}
```

Comment:
These props (aria-label, aria-labelledby, aria-describedby) are optional, but you aren't checking to see if they exist before using them?

Reply:
Sure - if these are null or undefined, they won’t get passed